nil pointer dereference when a node goes up

### What version of Cassandra are you using?

2.1.18

### What version of Gocql are you using?

Commit f52d33ca51e4216a6bf6af74f80e023e69700afd

### What did you do?

Restarted a node in the cluster.

### What did you expect to see?

No panic when the node goes up

### What did you see instead?

A nil pointer dereference:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6aff2d]

goroutine 11417 [running]:
foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*HostInfo).ConnectAddress(0x0, 0x0, 0x0, 0x0)
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/host_source.go:176 +0x4d
foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*ringDescriber).getHostInfo(0xc420532ec0, 0xc420b3a308, 0x4, 0x4, 0x2352, 0xc420b3a308, 0x4, 0x4)
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/host_source.go:597 +0x396
foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*Session).handleNewNode(0xc4200e0580, 0xc420b3a308, 0x4, 0x4, 0x2352, 0xc4200d2b01)
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/events.go:194 +0xda
foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*Session).handleNodeEvent(0xc4200e0580, 0xc422d14000, 0x1, 0x3e8)
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/events.go:162 +0x782
foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*Session).(foobar.com/foo/myapp/vendor/github.com/gocql/gocql.handleNodeEvent)-fm(0xc422d14000, 0x1, 0x3e8)
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/session.go:115 +0x48
created by foobar.com/foo/myapp/vendor/github.com/gocql/gocql.(*eventDebouncer).flush
	/go/src/foobar.com/foo/myapp/vendor/github.com/gocql/gocql/events.go:64 +0x68
```

I suspect the problem only appears when the cluster is misconfigured; one node from this cluster had a `peer` entry in `system.peers` which was an old IP address that no node had physically.
I also restarted nodes countless times before and never had this error.

I think I see what the problem is so I'll follow up with a PR

--- a/host_source.go
+++ b/host_source.go
@@ -632,7 +632,7 @@ func (r *ringDescriber) getHostInfo(ip net.IP, port int) (*HostInfo, error) {
 				return nil, err
 			}
 
-			if host.ConnectAddress().Equal(ip) {
+			if h.ConnectAddress().Equal(ip) {
 				host = h
 				break
 			}

